
#!/bin/bash

#PBS -l nodes=1:ppn=1
#PBS -l mem=30gb
#PBS -l walltime=23:59:00
#PBS -M z5434493@ad.unsw.edu.au
#PBS -m ae
 
#################### gapseq #################### 

# Version gapseq_v20210501 new system, new R lib directory.

module load perl/5.28.0
module load git/2.22.0
module load bedtools/2.27.1
module load blast+/2.9.0
module load hmmer/3.2.1
module load glpk/4.65
module load barrnap/0.9
module load gcc/7.3.0
module load exonerate/2.2.0
module load parallel/20190522
module load libsbml/5.18.0 # module load since 2020-12-29
# module add R/3.5.3 
module add R/3.6.1 # sybilSBML is necessary
# * DONE (CHNOSZ)
# * DONE (jsonlite)
module load cplex/12.9.0-academic  

# Navigate to where you want to run the job/store outputs 
cd /srv/scratch/z5434493/Ap_beatrix

# Pathway prediction 
/srv/scratch/z5434493/Softwares/gapseq/gapseq_20220825/gapseq find -p all -t Bacteria -b 200 -c 70 -l MetaCyc -y ./MAGs/AQ2.fasta 

#√   -p keywords such as pathways or subsystems (for example amino,nucl,cofactor,carbo,polyamine)
#√   -t Taxonomic range for sequences to be downloaded (default: Bacteria)
#√   -b Bit score cutoff for local alignment (default: 200)
#√   -n Consider superpathways of metacyc database
#√   -l Select the pathway database (MetaCyc(2712), KEGG(523), SEED(666), all(3922); default: metacyc,custom)


# Transporter prediction 
/srv/scratch/z5434493/Softwares/gapseq/gapseq_20220825/gapseq find-transport -b 100 -c 70 ./MAGs/AQ2.fasta

#√   -b bit score cutoff for local alignment (default: 50)
#√   -c coverage cutoff for local alignment (default: 75)

# Draft network reconstruction for gram neg bacteria
/srv/scratch/z5434493/Softwares/gapseq/gapseq_20220825/gapseq draft -r ./AQ2-all-Reactions.tbl -t ./AQ2-Transporter.tbl -p ./AQ2-all-Pathways.tbl -c ./AQ2.fasta -u 100 -l 50 -b neg

#√     -r|--blast.res          Blast-results table generated by gapseq.sh.
#√     -t|--transporter.res    Blast-results table generated by transporter.sh.
#√     -b|--biomass            Gram "pos" OR "neg" OR "archae" OR "auto"? Default: "auto". Please note: if set to "auto", the external programms barrnap, usearch, and bedtools are required.
#√     -u|--high.evi.rxn.BS    Reactions with an associated blast-hit with a bitscore above this value will be added to the draft model as core reactions (i.e. high-sequence-evidence reactions)
#√     -l|--min.bs.for.core    Reactions with an associated blast-hit with a bitscore below this value will be considered just as reactions that have no blast hit.

# Medium prediction
/srv/scratch/z5434493/Softwares/gapseq/gapseq_20220825/gapseq medium -m AQ2-draft.RDS -p AQ2-all-Pathways.tbl 

# Gapfill 
/srv/scratch/z5434493/Softwares/gapseq/gapseq_20220825/gapseq fill -m ./AQ2-draft.RDS -n AQ2-medium.csv -c ./AQ2-rxnWeights.RDS -g ./AQ2-rxnXgenes.RDS -b 50 -o ./2022_pred_diet_AQ2 -r TRUE

#√     -m|--model                  GapSeq-Draft-Model to be gapfilled (RDS or SBML)
#√     -n|--media                  tab- or komma separated table for media components. Requires three named columns: 1 - "compounds" (for metab. IDs), 2 - "name" (metab. name), 3 - "maxFlux" (maximum inflow flux)
#√     -c|--rxn.weights.file       Reaction weights table generated by gapseq function "generate_GSdraft.R" (RDS format).
#√     -g|--rxnXgene.table         Table with gene-X-reaction associations as generated by the "generate_GSdraft.R" (RDS format)
#√     -b|--bcore                  Minimum bitscore for reaction associated blast hits to consider reactions as core/candidate reactions. Default: 50
#√     -o|--output.dir             Directory to store results. Default: "gapfill".
#√     -r|--relaxed.constraints    Save final model as unconstraint network (i.e. all exchange reactions are open). Default: FALSE
